### PR TITLE
Fix external renderer

### PR DESCRIPTION
**Pull request by 6543**
as title

### CODE REVIEW

**Comment by 6543** [comment]
demos:
* https://codeberg-test.org/6543/reStructuredText_ReST
* https://codeberg-test.org/6543/asciidoc-test

File: modules/markup/external/external.go [hunk 1]
```diff
@@ -98,7 +98,14 @@ func (p *Renderer) Render(ctx *markup.RenderContext, input io.Reader, output io.
...
 
-	processCtx, cancel := context.WithCancel(ctx.Ctx)
+	parentCtx := context.Background()
+	if ctx != nil && ctx.Ctx != nil {
+		parentCtx = ctx.Ctx
+	} else {
+		log.Error("RenderContext did not provide context)")
+	}
+
+	processCtx, cancel := context.WithCancel(parentCtx)
```
Comment (zeripath) [inline modules/markup/external/external.go]: Not setting the renderctx is a fatal error and should not be tolerated.

Comment (6543) [reply modules/markup/external/external.go]: ok will return an error then :)

**Review by lunny — APPROVED** [review]

**Comment by noerw** [comment]
might be unrelated, but the demo links you provided give a 500 Internal Server Error

**Comment by 6543** [comment]
@noerw no its related, external renderer are broken, ot test clome mentioned repos to your test instamce and setup restructured text renderer. It should throw a 500, if you checkout this pull you should get a rendered page back

**Review by noerw — APPROVED** [review]

**Review by zeripath — APPROVED** [review]

**Pull request merged**